x/ecocredit/core/msg_mint_batch_credits_test.go

```diff
@@ -0,0 +1,40 @@
+package core
+
+import (
+	fmt "fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/thanhpk/randstr"
+)
+
+func TestMsgMintBatchCredits(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	issuer := genAddress()
+
+	tcs := []struct {
+		name string
+		m    MsgMintBatchCredits
+		err  string
+	}{
+		{"invalid issuer", MsgMintBatchCredits{Issuer: "invalid"}, "issuer"},
+		{"invalid denom", MsgMintBatchCredits{Issuer: issuer, BatchDenom: "XXX"}, "invalid denom"},
+		{"invalid note",
+			MsgMintBatchCredits{Issuer: issuer, BatchDenom: batchDenom, Note: randstr.String(514)}, "note must"},
+		{"missing origin tx",
+			MsgMintBatchCredits{Issuer: issuer, BatchDenom: batchDenom}, "origin_tx is required"},
+
+		{"good-no-note", MsgMintBatchCredits{Issuer: issuer, BatchDenom: batchDenom, OriginTx: &batchOrigTx, Issuance: batchIssuances}, ""},
+		{"good-note", MsgMintBatchCredits{Issuer: issuer, BatchDenom: batchDenom, OriginTx: &batchOrigTx, Note: randstr.String(300),
+			Issuance: batchIssuances}, ""},
```

Comment:
Might be nice to give each field it's own line to help with readability but ok to leave as is.

Reply:
Do you think it's not readable? I find it easier to read if it doesn't consume significant amount of vertical space.

Reply:
It is readable but I do find it easier to read if each field is on its own line and does not go beyond the line width. Not a big deal though. Maybe just personal preference. As mentioned, ok to leave as is.

```
{"good-no-note", MsgMintBatchCredits{Issuer: issuer, BatchDenom: batchDenom, OriginTx: &batchOrigTx, Issuance: batchIssuances}, ""},
```

vs

```
{
	"good-no-note",
	MsgMintBatchCredits{
		Issuer:     issuer,
		BatchDenom: batchDenom,
		OriginTx:   &batchOrigTx,
		Issuance:   batchIssuances,
	},
	"",
},
```

Reply:
stack many of them and your test will take multiple of screens.